quickwit/quickwit-common/src/lib.rs

```diff
@@ -52,6 +53,9 @@ pub use kill_switch::KillSwitch;
 pub use progress::{Progress, ProtectedZoneGuard};
 use tracing::{error, info};
 
+// TODO: remove `NON_ZERO_USIZE_MIN` once `NonZeroUsize::MIN` is stabilized.
+pub const NON_ZERO_USIZE_MIN: NonZeroUsize = unsafe { NonZeroUsize::new_unchecked(1) };
```

Comment:
@guilload  does it sound reasonable to use this const? It just avoid calling `NonZeroUsize::new(1).unwrap()` everywhere and we can replace it by `NonZeroUsize::MIN` later.

Reply:
What about a macro to perform the check at compile time?